NightscoutServiceKit/Extensions/DeviceStatus.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  DeviceStatus.swift
+//  NightscoutServiceKit
+//
+//  Created by Darin Krauss on 10/17/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+import LoopKit
+import NightscoutUploadKit
+
+extension DeviceStatus {
+
+    init(storedDosingDecision: StoredDosingDecision) {
+        var iob: IOBStatus?
+        if let insulinOnBoard = storedDosingDecision.insulinOnBoard {
+            iob = IOBStatus(timestamp: insulinOnBoard.startDate, iob: insulinOnBoard.value)
+        }
+
+        var cob: COBStatus?
+        if let carbsOnBoard = storedDosingDecision.carbsOnBoard {
+            cob = COBStatus(cob: carbsOnBoard.quantity.doubleValue(for: HKUnit.gram()), timestamp: carbsOnBoard.startDate)
+        }
+
+        var predicted: PredictedBG?
+        if let predictedGlucose = storedDosingDecision.predictedGlucose, let startDate = predictedGlucose.first?.startDate {
+            let values = predictedGlucose.map { $0.quantity }
+            predicted = PredictedBG(startDate: startDate, values: values)
+        }
+
+        var recommended: RecommendedTempBasal?
+        if let tempBasalRecommendationDate = storedDosingDecision.tempBasalRecommendationDate {
+            recommended = RecommendedTempBasal(timestamp: tempBasalRecommendationDate.date, rate: tempBasalRecommendationDate.recommendation.unitsPerHour, duration: tempBasalRecommendationDate.recommendation.duration)
+        }
+
+        var loopEnacted: LoopEnacted?
+        if case .some(.tempBasal(let tempBasal)) = storedDosingDecision.pumpManagerStatus?.basalDeliveryState /*, lastTempBasalUploaded?.startDate != tempBasal.startDate  TODO: How to address this? */ {
```

Comment:
[nit] should this comment be removed?

Reply:
Yes, per @ps2's comment below, will remove and create a JIRA ticket to handle the correct way. (This commented out code was copied from the old Nightscout uploader code that attempted to infer a new dosing decision. I think calling out a  new dosing decision explicitly will be a better solution.)